### PR TITLE
copy: compression: default to gzip

### PR DESCRIPTION
**Pull request by vrothberg**
Unless specified otherwise, default to using the gzip compressor.  If
left nil, we'll run into a nil dereference breaking existing consumers
of containers/image.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@giuseppe @mtrmac @rhatdan PTAL

That's fixing a regression and prevents consumers from breaking.

**Comment by rhatdan** [comment]
LGTM
Can we get a PR into skopeo to fix the test suite.

**Comment by vrothberg** [comment]
Skopeo PR https://github.com/containers/skopeo/pull/706

**Comment by giuseppe** [comment]
there is also https://github.com/containers/skopeo/pull/671 that is ready for review.  If there is nothing holding it, let's merge it so we can get zstd into Skopeo

**Comment by giuseppe** [comment]
The Skopeo PR passes the tests.  I'll merge

File: copy/copy.go [hunk 1]
```diff
@@ -181,10 +182,18 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
 		// FIXME? The cache is used for sources and destinations equally, but we only have a SourceCtx and DestinationCtx.
 		// For now, use DestinationCtx (because blob reuse changes the behavior of the destination side more); eventually
 		// we might want to add a separate CommonCtx — or would that be too confusing?
-		blobInfoCache: blobinfocache.DefaultCache(options.DestinationCtx),
-
-		compressionFormat: *options.DestinationCtx.CompressionFormat,
-		compressionLevel:  options.DestinationCtx.CompressionLevel,
+		blobInfoCache:    blobinfocache.DefaultCache(options.DestinationCtx),
+		compressionLevel: options.DestinationCtx.CompressionLevel,
```
Comment (mtrmac) [inline copy/copy.go]: `options.DestinationCtx` can be `nil`, per the common semantics of `*types.SystemContext`.

File: copy/copy.go [hunk 2]
```diff
@@ -181,10 +182,18 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
...
+	}
+	// Default to using gzip compression unless specified otherwise.
+	if options.DestinationCtx.CompressionFormat == nil {
```
Comment (mtrmac) [inline copy/copy.go]: `options.DestinationCtx` can be `nil`.

File: copy/copy.go (hunk 1 above)
Comment (vrothberg) [reply copy/copy.go]: @giuseppe, can you take care of the remaining clean ups for that? And line 189?

File: copy/copy.go (hunk 2 above)
Comment (vrothberg) [reply copy/copy.go]: https://github.com/containers/image/pull/688

Comment (vrothberg) [reply copy/copy.go]: https://github.com/containers/image/pull/688

**Pull request merged**